library/Fission/User/Mutation.hs

```diff
@@ -64,7 +64,12 @@ create username password email herokuAddOnId now =
         , userModifiedAt    = now
         }
 
-      return (Right userID)
+      insertUnique newUserRecord >>= \case
+        Just userID ->
+          return (Right userID)
+
+        Nothing ->
+          return (Left Error.AlreadyExists)
```

Comment:
I like this phrasing! It's very clear. Just for science, here's another phrasing:

```haskell
insertUnique newUserRecord >>= return . maybe (Left Error.AlreadyExists) Right
```

🤔

Hmm, yeah, I like the `case` 👍